ci/nightly/Jenkinsfile

```diff
@@ -0,0 +1,71 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+
+@Library('hibernate-jenkins-pipeline-helpers@1.17') _
+
+def withMavenWorkspace(Closure body) {
+	withMaven(jdk: 'OpenJDK 21 Latest', maven: 'Apache Maven 3.9',
+			mavenLocalRepo: env.WORKSPACE_TMP + '/.m2repository',
+			options: [
+					// Artifacts are not needed and take up disk space
+					artifactsPublisher(disabled: true),
+					// stdout/stderr for successful tests is not needed and takes up disk space
+					// we archive test results and stdout/stderr as part of the build scan anyway,
+					// see https://ge.hibernate.org/scans?search.rootProjectNames=Hibernate%20Search
+					junitPublisher(disabled: true)
+			]) {
+		withCredentials([string(credentialsId: 'ge.hibernate.org-access-key',
+				variable: 'DEVELOCITY_ACCESS_KEY')]) {
+			withGradle {
+				// withDevelocity, actually: https://plugins.jenkins.io/gradle/#plugin-content-capturing-build-scans-from-jenkins-pipeline
+				body()
+			}
+		}
+	}
+}
+
+pipeline {
+	agent none
+	triggers {
+		cron '@midnight'
+	}
```

Comment:
We'll need to make sure the job created through the Jenkins web UI excludes pull requests and dependabot branches, otherwise this cron will be very expensive.

Reply:
Yeah... I guess we should target `main` only at this time and 7.2+ branches.. any other branches will fail the check

Reply:
Older branches are not an issue: they don't have the Jenkinsfile so won't get a run anyway.